File: app/src/main/java/br/com/jonathanzanella/myexpenses/expense/ExpenseContract.java
package br.com.jonathanzanella.myexpenses.expense;

import android.content.Context;
import android.support.annotation.StringRes;

import org.joda.time.DateTime;

import br.com.jonathanzanella.myexpenses.bill.Bill;
import br.com.jonathanzanella.myexpenses.chargeable.Chargeable;
import br.com.jonathanzanella.myexpenses.validations.ValidationError;

interface ExpenseContract {
	interface View {
		Context getContext();
		void setTitle(@StringRes int string);
		void setTitle(String string);
		void showExpense(Expense expense);
	}

	interface EditView extends View {
		Expense fillExpense(Expense expense);
		void finishView();
		void showError(ValidationError error);
		int getInstallment();
		int getRepetition();
		void onDateChanged(DateTime date);
		void onBillSelected(Bill bill);
		void onChargeableSelected(Chargeable chargeable);
	}
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/expense/ExpenseContract.kt
package br.com.jonathanzanella.myexpenses.expense

import android.content.Context
import android.support.annotation.StringRes
import br.com.jonathanzanella.myexpenses.bill.Bill
import br.com.jonathanzanella.myexpenses.chargeable.Chargeable
import br.com.jonathanzanella.myexpenses.validations.ValidationError
import org.joda.time.DateTime

interface ExpenseContract {
    interface View {
        val context: Context
        fun setTitle(@StringRes string: Int)
        fun setTitle(string: String)
        fun showExpense(expense: Expense)
    }

    interface EditView : View {
        fun fillExpense(expense: Expense): Expense
        fun finishView()
        fun showError(error: ValidationError)
        fun onDateChanged(date: DateTime)
        fun onBillSelected(bill: Bill?)
        fun onChargeableSelected(chargeable: Chargeable)
    }
}
